File: loan/src/main/java/com/binance4j/loan/client/LoanMapping.java
package com.binance4j.loan.client;

import java.util.List;
import java.util.Map;

import com.binance4j.core.client.RestMapping;
import com.binance4j.loan.dto.LoanIncome;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.QueryMap;

/** {@link LoanClient} mapping. */
public interface LoanMapping extends RestMapping {
	/** The base uri. */
	String BASE = "/sapi/v1/loan/";

	/**
	 * @param map Query map.
	 * @return The generated Retrofit call.
	 */
	@GET(BASE + "income")
	@Headers({ SIGNED_H, "X-WEIGHT: 6000", UID_H })
	Call<List<LoanIncome>> getLoansIncome(@QueryMap Map<String, Object> map);
}

File: connectors/src/main/kotlin/com/binance4j/loan/client/LoanMapping.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.binance4j.loan.client

import com.binance4j.core.Binance4j.SIGNED_H
import com.binance4j.core.Binance4j.UID_H
import com.binance4j.loan.dto.LoanIncome
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.QueryMap

/**
 * [LoanClient] mapping.
 */
interface LoanMapping {
	/**
	 * @param map Query map.
	 * @return The generated Retrofit call.
	 */
	@GET("/sapi/v1/loan/income")
	@Headers(SIGNED_H, "X-WEIGHT: 6000", UID_H)
	@JvmSuppressWildcards
	fun getLoansIncome(@QueryMap map: Map<String, Any>): Call<List<LoanIncome>>
}